src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -246,10 +237,56 @@ func (s *commitLogSource) Read(
 		zap.Duration("took", s.nowFn().Sub(startSnapshotsRead)))
 	span.LogEvent("read_snapshots_done")
 
+	commitLogResult, err := s.readCommitLog(namespaces, span)
```

Comment:
unfortunately the diff is not great. I essentially moved all of the commit log reading logic into a new method `readCommitLog`. additionally I changed the `for` loop below to iterate over the provided namespaces and _not_ the namespace results from the commit log. didn't seem necessary to use the results in the for loop and it allowed the results to be private in the new method.